src/DataValidation.py

```diff
@@ -9,6 +9,8 @@ class DataValidation():
     game_table = {}
     item_table = []
     location_table = []
+    location_events_table = []

```

Comment:
This doesn't belong in DataValidation. You're not validating anything with it. 

If you need it to be shared for multiple places, it should go on the World itself. And you can build the list in the Regions file.

Reply:
I see what you're saying. You need to read the event location names from the requires to know which event locations to create.

Problem with putting them in DataValidation is that it shouldn't do anything atomic. Shouldn't store anything for reuse, shouldn't construct anything, change anything... It just should look at the data that's there and detect issues. We don't want to end up tracking down issues in the validation causing bugs in generation, because that would be kinda cursed. 😃 Also, if we ever make validation optional, it shouldn't break generation.

But it's okay to just use list comprehension on the location_table to get locations that have event locations, then iterate those in a new loop. Can't imagine that would have much impact, and makes more sense to move event location creation / prep closer to Regions like you wanted to.

In fact, since it's going to end up on the world object, I could also see the list happening in Data.py and getting imported by the world. Then, in the future, we could run validation on the event location names if we wanted to. But that's not necessary for this. Just somewhere more appropriate than DataValidation, and a new loop is fine. 😃

Reply:
your validation is already optional if you generate with skip output 😉

Reply:
Why tho? You went from using an existing loop in DataValidation that was doing something else but also did what you wanted, to not using a loop at all and just creating unnecessary event locations. I'm not understanding why you can't just make that same loop again in one of the proper places.